dqgen/services/base_query_generator.py

```diff
@@ -7,22 +7,30 @@
 
 """ """
 import abc
-import pathlib
 
 
 class BaseQueryGenerator(abc.ABC):
```

Comment:
This calss can become concrete, and take the template as a parameter when instantiated.